### PR TITLE
Add DesignatedVoting contract for two-key voting system

### PR DESCRIPTION
**Pull request by ptare**
Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: core/contracts/DesignatedVoting.sol [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @notice Forwards a reveal to Voting.
+     */
+    function revealVote(bytes32 identifier, uint time, int price, int salt) external onlyRoleHolder(uint(Roles.Voter)) {
+        _getVotingAddress().revealVote(identifier, time, price, salt);
+    }
+
+    /**
+     * @notice Forwards a reward retrieval to Voting.
+     */
+    function retrieveRewards() external onlyRoleHolder(uint(Roles.Voter)) {
```
Comment (mrice32) [inline core/contracts/DesignatedVoting.sol]: Should we add a method that allows the Voter to call `migrateTokens` on the `TokenMigrator` contract? If we do, we could either add the `TokenMigrator` to the `Finder` or we could allow the Voter to pass in an address (may have negative security implications).

WDYT?

File: core/contracts/DesignatedVoting.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    // Reference to the UMA Finder contract, allowing Voting upgrades to be performed without requiring any calls to
+    // this contract.
+    Finder private finder;
+
+    constructor(address finderAddress) public {
+        _createExclusiveRole(uint(Roles.Owner), uint(Roles.Owner), msg.sender);
+        _createExclusiveRole(uint(Roles.Voter), uint(Roles.Owner), msg.sender);
+        // TODO(ptare): In this contract, a separate withdraw role doesn't make sense. Our choices are to:
+        // 1. Add a method to Withdrawable `setWithdrawRole()` to allow setting the owner as the withdrawer.
```
[the remark this answers is not in the document]
Comment (ptare) [reply core/contracts/DesignatedVoting.sol]: Done.

File: core/contracts/DesignatedVoting.sol (hunk 1 above)
Comment (ptare) [reply core/contracts/DesignatedVoting.sol]: We could, but I thought we allowed anyone to call the TokenMigrator on anyone's behalf? So would the migrate call need to go through this contract? (I may have forgotten how this works though)

Comment (mrice32) [reply core/contracts/DesignatedVoting.sol]: Oh! You're totally right - nice catch. WAI :)

**Pull request merged**